libs/sdk-backend-tiger/src/convertors/fromBackend/afm/GrandTotalsConverter.ts

```diff
@@ -0,0 +1,203 @@
+        // If the grand total belongs to multiple or zero dimensions it is either
+        // a grand total from a >2D result or it is a total of totals,
+        // neither of which can currently be handled
+        return [];
+    }
+    // Here we assume that there are only two dimensions in the definition, as we need to find on which dimension
+    // the totals in this grand total result were defined. If there were more than two dimensions, then a total
+    // definition on a single dimension would result in >1 dimensions in `totalDimensions`, which we checked already.
+    // TODO revisit when multidimensional totals are supported
+    const dimensionIdx = definition.dimensions.findIndex((_, idx) => idx !== totalDimensions[0]);
+    const totalDefinitions = definition.dimensions[dimensionIdx]?.totals;
+    if (totalDefinitions === undefined) return [];
+
+    const transformedTotals: Record<number /* total index */, DataValue[]> = {};
+    const resultData = grandTotal.data as ResultData;
+    walk(resultData);
+    return Object.keys(transformedTotals)
+        .map(Number)
+        .map((totalIdx): GrandTotal => {
+            const data = transformedTotals[totalIdx];
+            return {
+                dimensionIdx,
+                totalIdx,
+                data,
+            };
+        });
+
+    /**
+     * Iteratively walk through the data, get headers at each point and use them to assign data values to a total.
+     */
+    function walk(data: ResultData, coordinates: number[] = []) {
+        data.forEach((dataValue: DataValue | ResultData, idx: number) => {
+            const position = [...coordinates, idx];
+            if (Array.isArray(dataValue)) {
+                walk(dataValue, position);
+            } else {
+                const headers = getHeadersAtPosition(headerItems, position);
+                const { totalType, measureIdentifier } = getTotalInfoFromHeaders(headers, definition);
+                if (!totalType) {
+                    // Cannot assign to a total without a total header
+                    return;
+                }
+                const totalIdx = getTotalIndex(totalDefinitions, totalType, measureIdentifier);
+                if (totalIdx === undefined) return;
+                const total = (transformedTotals[totalIdx] = transformedTotals[totalIdx] ?? []);
+                total.push(dataValue);
+            }
+        });
+    }
+}
+
+function getHeadersAtPosition(
+    headerItems: Readonly<IResultHeader[][][]>,
+    position: number[],
+): IResultHeader[] {
+    return (
+        position
+            .map((coord, dimIdx) => {
+                // header groups in dimension `dimIdx`
+                const headerGroups = headerItems[dimIdx];
+                // get header at coordinate `coord` from each group
+                return headerGroups.map((group) => group[coord]);
+            })
+            // flatten
+            .reduce((prev, curr) => prev.concat(curr))
```

Comment:
You could simplify this using the [flatMap](https://lodash.com/docs/4.17.15#flatMap) function from lodash